backends/projects/client.go

```diff
@@ -0,0 +1,153 @@
+package projects
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/ugent-library/biblio-backoffice/models"
+	"github.com/ugent-library/projects-service/api/v1"
+)
+
+type Config struct {
+	APIUrl string
+	APIKey string
+}
+
+type securitySource struct {
+	apiKey string
+}
+
+func (s *securitySource) ApiKey(ctx context.Context, operationName string) (api.ApiKey, error) {
+	return api.ApiKey{APIKey: s.apiKey}, nil
+}
+
+type Client struct {
+	config Config
+	client *api.Client
+}
+
+func New(c Config) (*Client, error) {
+	client, err := api.NewClient(c.APIUrl, &securitySource{c.APIKey})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		config: c,
+		client: client,
+	}, nil
+}
+
+func (c *Client) GetProject(id string) (*models.Project, error) {
+	ctx := context.TODO()
+
+	res, err := c.client.GetProject(ctx, &api.GetProjectRequest{ID: fmt.Sprintf("urn:iweto:%s", id)})
+	if err != nil {
+		return nil, err
+	}
+
+	switch ap := res.(type) {
+	case *api.GetProject:
+		return c.mapper(ap), nil
+	case *api.ErrorStatusCode:
+		return nil, fmt.Errorf("projects suggestions: project service has thrown an error: code: %d, id: %s", ap.GetStatusCode(), id)
+	}
+
+	return nil, fmt.Errorf("projects suggestions: something went wrong (id: %s)", id)
+}
+
+func (c *Client) SuggestProjects(q string) ([]models.Project, error) {
+	ctx := context.Background()
+
+	res, err := c.client.SuggestProjects(ctx, &api.SuggestProjectsRequest{Query: q})
+	if err != nil {
+		return nil, err
+	}
+
+	projects := make([]models.Project, 0)
```

Comment:
this better written as:

```
        projects := make([]models.Project, len(res.Data))

	for i, hit := range res.Data {
		p := c.mapper(&hit)
		projects[i] = *p
	}
```

much more efficient

you can gain an extra line if you omit the temporary `p` variable

Reply:
done. temp var is a pointer variable; hard to omit since we can't directly dereference a function pointer return.